examples/quantum_associative_memory.cpp

```diff
@@ -0,0 +1,73 @@
+//////////////////////////////////////////////////////////////////////////////////////
+//
+// (C) Daniel Strano and the Qrack contributors 2017-2019. All rights reserved.
+//
+// This example demonstrates an example of a "quantum associative memory" network with the Qrack::QNeuron
+// class. QNeuron is a type of "neuron" that can learn and predict in superposition, for general machine learning
+// purposes.
+//
+// Licensed under the GNU Lesser General Public License V3.
+// See LICENSE.md in the project root or https://www.gnu.org/licenses/lgpl-3.0.en.html
+// for details.
+
+#include <iostream> // For cout
+
+// "qfactory.hpp" pulls in all headers needed to create any type of "Qrack::QInterface."
+#include "qfactory.hpp"
+// "qneuron.hpp" defines the QNeuron class.
+#include "qneuron.hpp"
+
+using namespace Qrack;
+
+int main()
+{
+    const bitLenInt InputCount = 4;
+    const bitLenInt OutputCount = 4;
+    const bitCapInt InputPower = 1U << InputCount;
+    // const bitCapInt OutputPower = 1U << OutputCount;
+    const real1 eta = 1.0;
+
+#if ENABLE_OPENCL
```

Comment:
You know, I really felt like we had some default QINTERFACE options so you didn't have to have ENABLE_OPENCL out at the leaf level.  Maybe that was just in my head...

Reply:
We do! I replaced this in both examples.